tree/30_generic_methods/variable_dict_merge.cf

```diff
@@ -0,0 +1,97 @@
+#####################################################################################
+# Copyright 2017 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Variable dict merge
+# @description Define a variable resulting of the merge of two other variables
+# @documentation To use the generated variable, you must use the form `${variable_prefix.variable_name[key]}` with each name replaced with the parameters of this method.
+#
+# The resulting variable will be the merge of the two parameters, which means it is built by:
+#
+# * Taking the content of the first variable
+# * Adding the content of the second variable, and replace the kays that were already there
+#
+# It is only a one-level merge, and the value of the first-lvel key will be completely replaced by the merge.
+#
+#### Usage
+#
+# If you have a `prefix.variable1` variable defined by:
+#
+# ```json
+# { "key1": "value1", "key2": "value2", "key3": { "keyx": "valuex" } }
+# ```
+#
+# And a `prefix.variable2` variable defined by:
+#
+# ```json
+# { "key1": "different", "key3": "value3", "key4": "value4" }
+# ```
+#
+# And that you use:
+#
+# ```
+# variablr_dict_merge("prefix", "variable3, "prefix.variable1", "prefix.variable2")
+# ```
+#
+# You will get a `prefix.variable3` variable containing:
+#
+# ```json
+# {
+#   "key1": "different",
+#   "key2": "value2",
+#   "key3": "value3",
+#   "key4": "value4"
+# }
+# ```
+#
+# @parameter variable_prefix The prefix of the variable name
+# @parameter variable_name   The variable to define, the full name will be variable_prefix.variable_name
+# @parameter first_variable  The first variable, that will be overriden if necessary (in the form variable_prefix.variable_name)
```

Comment:
Could you use the word source instead of overridden to avoid confusion about the state of the variable after the call